src/jllvm/class/ClassFile.hpp

```diff
@@ -343,6 +357,13 @@ class FieldInfo
     AttributeMap m_attributes;
 
 public:
+    FieldInfo() = default;
+    ~FieldInfo() = default;
+    FieldInfo(const FieldInfo&) = delete;
+    FieldInfo& operator=(const FieldInfo&) = delete;
+    FieldInfo(FieldInfo&&) = default;
+    FieldInfo& operator=(FieldInfo&&) = default;
+
```

Comment:
Why are these needed? Aren't theses redundant? Ditto with the `MethodInfo` below. 
If the default constructor is just added to make it possible to use `std::generate` I'd rather keep the code as is (although adding `reserve` calls makes sense).
Alternatively you can even use `std::generate_n` with a `std::back_inserter` if you insist on getting rid of the `for` loops.